### ISSUE TITLE
HYPERSPACE warpin type collision NaN

### ISSUE THREAD

**Issue by Droid803** [scp-fs2open/fs2open.github.com#1719]
In certain conditions, when a ship arriving via HYPERSPACE warp in type collides with another ship results in a NaN speed vector (and sometimes a NaN distance as well). While this can be avoided in most cases simply by placing arriving ships far enough apart, there's always the chance that a ship arriving with the "Near Ship" arrival anchor randomly selects an arrival vector that causes it to hit another ship, which results in a NaN (along with it's associated problems on the screen going black and whatnot)

Please see attached sample mission (run in DE v1.1b). Alt-X to summon a ship to hit the one in the mission.

[NaN_SpeedVec_DE11b.zip](https://github.com/scp-fs2open/fs2open.github.com/files/1958511/NaN_SpeedVec_DE11b.zip)

**Comment by Droid803** [comment]
Some examples of what happens:

https://cdn.discordapp.com/attachments/223511391531827200/439597409300119563/screen0136.png
https://cdn.discordapp.com/attachments/223511391531827200/439597431882121216/screen0138.png

**Comment by Goober5000** [comment]
This should be fixable by setting the arrival stage so that the ship doesn't count as "arrived" until it has stopped hyperspacing.  If necessary, an intermediate arrival stage (e.g. WARPING_BUT_NOT_ARRIVED) should be inserted between existing ones.

Turning off collision detection will fix the immediate issue, but won't fix the underlying issue that also affects AI, pathing, and physics.

**Comment by Goober5000** [comment]
Actually I took a look and I think this should be very simple to fix.  @ln-zookeeper put his hyperspace hack in the wrong location.  A few tweaks to `Arriving_stage_1` and it should work properly.

@Droid803 what do I need to download to run your test mission?

**Comment by Goober5000** [comment]
I took another look and maybe @ln-zookeeper's hack was the right thing to do after all.  Still awaiting feedback from @Droid803 on both the requirements to run the test mission and his experience using the lightweight mod with the latest nightly. :sigh:

Another assumption was that the line
```
pair->next_check_time = timestamp(0);	// check next time
```
in `collide_ship_ship` should be changed so that, if the ship is hyperspacing, the timestamp is the time until stage 1 finishes.  But it looks like the existing line achieves the same effect anyhow, because the collision check seems to be skipped until stage 1 is actually finished.

**Comment by niffiwan** [comment]
[bbb-shp.tbm.txt](https://github.com/scp-fs2open/fs2open.github.com/files/2897430/bbb-shp.tbm.txt)
[NaN_SpeedVec.fs2.txt](https://github.com/scp-fs2open/fs2open.github.com/files/2897431/NaN_SpeedVec.fs2.txt)

Github brought this issue to my attention. I tried using the sample mission to repro with DE 1.2.9 on knossos; but hitme didn't hit realhard :(

I also converted the mission to work with FS2; but the bug didn't reappear, i.e. hitme hit realhard, but no NaN occurred. Just a spinning Deimos & a dead Asmodeus (clobbered by the spinning Deimos)

Tested with nightly from 5th Feb 2019.

Help @Droid803 !
